api_server.py

```
@@ -549,4 +563,9 @@ def add_security_headers(response):
 """
 
 if __name__ == "__main__":
-    app.run(host="0.0.0.0", port=int(os.environ.get("PORT", 5000)))
+    # Prevent running development server in production
+    if os.environ.get("FLASK_ENV") == "production":
+        print("Error: app.run() is disabled in production. Use gunicorn instead.")
+        os._exit(1)
+    # Binding to 0.0.0.0 is intended for containerized deployment
+    app.run(host="0.0.0.0", port=int(os.environ.get("PORT", 5000)))  # nosec B104
```

Ruff (0.15.0)

[error] Possible binding to all interfaces

(S104)